clients/java-logger/shell/src/main/java/com/abcxyz/lumberjack/loggingshell/LoggingShellApplication.java

```diff
@@ -41,22 +46,30 @@ public static void main(String[] args) throws IOException {
   static class JWKHandler implements HttpHandler {
     // Matching private key here:
     // https://github.com/abcxyz/lumberjack/blob/main/integration/testrunner/grpcrunner/grpc.go#L59
-    private static final String PUBLIC_JWK =
-        "{"
-            + "\"crv\": \"P-256\","
-            + "\"kid\": \"integ-key\","
-            + "\"kty\": \"EC\","
-            + "\"x\": \"hBWj8vw5LkPRWbCr45k0cOarIcWgApM03mSYF911de4\","
-            + "\"y\": \"atcBji-0fTfKQu46NsW0votcBrDIs_gFp4YWSEHDUyo\""
-            + "}";
+    private static String parsePublicKey() throws Exception {
+      JSONParser parser = new JSONParser();
+      try {
+        Object obj = parser.parse(new FileReader("/etc/lumberjack/public_key.json"));
+        JSONObject jsonObject = (JSONObject) obj;
+        String decoded = (String) jsonObject.get("decoded");
+        return decoded;
+      } catch (Exception e) {
+        throw e;
+      }
+    }
 
     @Override
     public void handle(HttpExchange t) throws IOException {
```

Comment:
This is duplicated from the one in Java gRPC. We could deduplicate them. Could be a follow up.

Reply:
Will do them as a follow up.